client/rest-high-level/src/test/java/org/elasticsearch/client/ml/datafeed/DatafeedConfigTests.java

```diff
@@ -103,6 +103,13 @@ public static DatafeedConfig createRandom() {
         if (randomBoolean()) {
             builder.setChunkingConfig(ChunkingConfigTests.createRandomizedChunk());
         }
+        if (randomBoolean()) {
```

Comment:
nit: My understanding is that you can have a time value (`setDelayedDataCheckWindow` ) set even if `setShouldRunDelayedDataCheck == false` because the check can be toggled on or off and that's the reason for using 2 fields rather than one. If that is the case the test should reflect that and remove the condition `if (shouldRunDelayedDataCheck) { // set the time window...`

Reply:
Same with DatafeedUpdateTests

Reply:
you are correct. I shall update the tests